### ISSUE TITLE
Pyplot hlines and vlines do not use the 'lines.color' property in rcParams by default

### ISSUE THREAD

**Issue by maij** [matplotlib/matplotlib#16482]
### Bug report

**Bug summary**

`hlines` and `vlines` color does not default to `lines.color` in rcParams. The default argument is 'k' which doesn't work with dark backgrounds.

**Code for reproduction**

```python
import matplotlib.pyplot as plt
import matplotlib as mpl
plt.style.use('dark_background')

plt.figure()
with mpl.rc_context({'lines.color':'white'}):
   plt.hlines(0.5, 0, 1) # defaults to a black line, which doesn't work with a dark background
   plt.vlines(0.5, 0, 1, colors=None) # This will fall back to `lines.color`
```

**Actual outcome**

![image](https://user-images.githubusercontent.com/7358024/74382315-4f63c200-4e41-11ea-967f-bd27ab6e78d8.png)

**Expected outcome**

![image](https://user-images.githubusercontent.com/7358024/74382350-62769200-4e41-11ea-949b-c36e6013af21.png)

**Matplotlib version**

  * Operating system: Windows 10
  * Matplotlib version: 3.1.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): module://ipykernel.pylab.backend_inline
  * Python version: 3.7.4
  * Jupyter version (if applicable):
  * Other libraries:

**Comment by RiazCharania** [comment]
Is anyone else currently working on this? If not I'll give it a try.

**Comment by JacobChamberlain** [comment]
We would like to work on this issue if you have not started yet.

**Comment by maij** [comment]
You're more than welcome to get started on this! I haven't started yet.

**Comment by RiazCharania** [comment]
> We would like to work on this issue if you have not started yet.

My team has already started on it, sorry.


> You're more than welcome to get started on this! I haven't started yet.

Thanks for the confirmation!

**Comment by ImportanceOfBeingErnest** [comment]
There is discussion in https://github.com/matplotlib/matplotlib/issues/16714 about whether `axhline` should use the grid's color or spline color. I suppose that would apply to `vlines/hlines` as well. So probably this needs more discussion.

**Comment by timhoffm** [comment]
> I suppose that would apply to vlines/hlines as well.

Not necessarily. `axhline` is typically used for marking single special values (e.g. https://matplotlib.org/devdocs/gallery/pyplots/axline.html). Whereas `hlines` are more data like, so it might make sense to have them use the color cycle, but that of course, would be a breaking change, which is probably only viable for 4.0.

**Comment by timhoffm** [comment]
Closed by #16953.
